tests/Authentication/LoginTest.php

```diff
@@ -82,4 +82,88 @@ public function testAuthenticateClient()
         self::assertGreaterThan($expires - 2, $expiry);
         self::assertLessThan($expires + 2, $expiry);
     }
+
+    public function testAuthenticateClientWithFunctionInApiRequestHeaders()
+    {
+        $guzzle = new Client(['base_url' => 'http://example.com/api']);
+        $guzzle->setDefaultOption('headers', ['X-Test' => '1']);
+
+        $expiresIn = 5000;
+        $expires = time() + $expiresIn;
+        $mock = new Mock([
+            new Response(200, [], Stream::factory(json_encode((object) [ // auth
+                'headerTokens' => [
+                    'token' => 1234
+                ],
+                'xTestToken' => 5678,
+                'queryToken' => 4321,
+                'expiresIn' => $expiresIn,
+            ]))),
+            new Response(200, [], Stream::factory(json_encode((object) [ // api call
+                'data' => [1,2,3]
+            ]))),
+            new Response(200, [], Stream::factory(json_encode((object) [ // api call
+                'data' => [1,2,3]
+            ])))
+        ]);
+        $guzzle->getEmitter()->attach($mock);
+
+        $history = new History();
+        $guzzle->getEmitter()->attach($history);
+
+        $restClient = new RestClient($guzzle, new NullLogger());
+
+        $attrs = ['first' => 1, 'second' => 'two'];
+
+        $api = [
+            'loginRequest' => [
+                'endpoint' => 'login',
+                'params' => ['par' => ['attr' => 'first']],
+                'headers' => ['X-Header' => ['attr' => 'second']],
+                'method' => 'POST'
+            ],
+            'apiRequest' => [
+                'headers' => [
+                    'X-Test-Auth' => 'xTestToken',
+                    'Authorization' => [
+                        'function' => 'concat',
+                        'args' => [
+                            'Bearer',
+                            ' ',
+                            'bearerToken' => 'headerTokens.token'
```

Comment:
tu je to zrovna jedno co to je ... a key `bearerToken`  mozem premenovat na `whatever`, podstatne je, ze je to "key"=>"val" a val hovori, odkial sa zoberu data ... nedal som tam attr aby sa to neplietlo s tym, ked sa attr bere z config